R/060_utils.R

```diff
@@ -0,0 +1,348 @@
+                                     stringr::str_remove_all(" _ "), character)) %>%
+      dplyr::ungroup() %>%
+      dplyr::arrange(h_border_group, row, col) %>%
+      dplyr::select(row, col, character = value)
+    
+    header_df <-
+      header_df %>%
+      dplyr::select(-character) %>%
+      dplyr::left_join(filled_join, by = c("row", "col"))
+  }
+  
+  header_df
+}
+
+
+
+#' Unmerge and fill in blanks for headers
+#'
+#' This is an internal function that function ensures that merged cells are unmerged.
+#'
+#' @param header_df a data frame containing header cells.
+#' @param header_fill he criteria according to which blank cells will be filled.
+#' @param formats the formats of the workbook associated with the tidyxl data frame.
+#' @param direction the direction in which headers are filled.
+
+fill_blanks_in_headers <- function(header_df, header_fill, formats, direction) {
+  if (direction %in% c("N", "S", "up", "down")) {
+    header_df <- suppressMessages(fill_blanks_in_col_headers(header_df, header_fill, formats))
+  } else if (direction %in% c("W", "E", "left", "right")) {
+    header_df <- suppressMessages(fill_blanks_in_row_headers(header_df, header_fill, formats))
+  } else {
+    stop("Please check the direction you are providing")
+  }
+
+  header_df
+}
+
+#' Add a set of variables based on a list of formulae
+#'
+#' This is an internal function that takes a data frame and a list of formulae and appends an arbitrary number of columns to the data frame based on the formulae.
+#'
+#' @param df a data frame containing header cells.
+#' @param form_list list of formulae
+#' @param format the formats associated with the workbook containing the header_df cells.
+#'
+
+reduce_mutated <- function(df, form_list, format) {
+  current_quosure <- form_list[[1]]
+  var_name_sym <- rlang::sym(form_list[[2]])
+
+  df %>%
+    dplyr::mutate(!!var_name_sym := !!current_quosure)
+}
+
+#' Give quosure a name
+#'
+#' This is an internal function that adds a prefixed name to a quosure so that variables added to a data frame using this quosure have a predictable name.
+#' @param x quosure
+#' @param prefix prefix to be added in name
+#'
+
+append_name_to_quosure <- function(x, prefix = "grp_") {
+  list(
+    x,
+    paste0(
+      prefix,
+      x %>% rlang::as_label() %>% make.names() %>%
+        stringr::str_replace_all("\\.+", ".") %>% stringr::str_remove_all("(\\.$)|(^X\\.)") %>%
+        stringr::str_replace_all("\\.", "_") %>%
+        ifelse(stringr::str_sub(., start = 1, 1) %in% as.character(0:9), paste0("x", .), .)
+    )
+  )
+}
+
+#' Add a variable to a tidyxl data frame if that variable is missing
+#'
+#' This is an internal function that takes a tidyxl data frame and a variable name to be added if they are missing.
+#' @param sheet sheet nominated for tidying
+#' @param var name of the var to be added as a character string.
+#'
+add_variable_if_missing <- function(sheet, var) {
```

Comment:
Here's another way that might be faster

```r
bind_rows(mtcars, tibble(newcol = logical()))
```

Reply:
Sounds good to me!

Reply:
Addressed in 36a3c5833536aeba36beaac8344850d44a9eaf52. 
Had to use character because these columns are later filled with string values.